docs/new-resource.md

```diff
@@ -1,149 +1,215 @@
 }
-appendValueIntoMap(results, aws.AwsS3BucketResourceType, s3Bucket)
 ```
 
-- an instance of the cloud provider SDK that you will use to retrieve resources list
+So as you can see this contains the logic to create the github client (it might be created outside the repository if it
+makes sense to share it between multiple repositories). It also get a cache so every request is cached.
+Driftctl sometimes needs to retrieve list of resources more than once, so we cache every request to avoid unnecessary call.
+
+### Enumerator
 
-### Deserializer
+This is used to build a resources list. Enumerators can be found in `pkg/remote/<providername>/<type>_enumerator.go`. It will call the cloud provider SDK to get the list of resources.
 
-The deserializer is used when reading resource from the terraform provider or from the state.
-The interface contains a `Deserialize(values []cty.Value) ([]resource.Resource, error)` method that you'll implement.
+Note that at this point resources should not be entirely fetched.
+Most of the resource returned by enumerator have empty attributes: they only represent type and terraform id.
 
-You should then deserialize the obtained cty values into your resource and return the list.
+**There are exception to this**:
+- Sometime, you will need some more information about resources to retrieve them using the provider they should be added to the resource attribute maps.
+- For some more complex cases, middleware needs more information that the id and type and in order to make classic run of driftctl coherent with a run with deep mode activated,
+these informations should be fetched manually by the enumerator using the remote sdk.
 
-Example: [aws_s3_bucket_deserializer.go](https://github.com/cloudskiff/driftctl/blob/main/pkg/resource/aws/deserializer/s3_bucket_deserializer.go)
+Note that we use the classic repository to hide calls to the provider sdk.
+You will probably need to at least add a listing function to list you new resource.
 
-## Adding your resource
+You should use an already implemented Enumerator as example.
 
-There are two files you are going to edit to make driftctl aware of your new resource.
+For example when implementing ec2_instance resource you will need to add a ListAllInstances() function to `repository.EC2Repository`.
+It will be called by the enumerator to retrieve the instances list.
 
-For the state reader you will need to add your `CTYDeserializer` implementation into `iac/deserializers.go`.
-Just add an instance in the list:
+Enumerator constructor could use these arguments:
+- an instance of `Repository` that you will use to retrieved information about the resource
+- the global resource factory that should always be used to create a new `resource.Resource`
+
+Enumerator then need to implement:
+- `SupportedType() resource.ResourceType` that will return the constant you defined in the type file at first step
+- `Enumerate() ([]*resource.Resource, error)` that will return the resource listing.
```

Comment:
Remove the point at the end and change it to `that will return the list of resources`